cocotb/drivers/amba.py

```diff
@@ -26,29 +26,54 @@
 
 """Drivers for Advanced Microcontroller Bus Architecture."""
 
+import array
+import enum
+
 import cocotb
-from cocotb.triggers import RisingEdge, ReadOnly, Lock
-from cocotb.drivers import BusDriver
 from cocotb.binary import BinaryValue
-
-import array
+from cocotb.drivers import BusDriver
+from cocotb.triggers import ClockCycles, Combine, Lock, ReadOnly, RisingEdge
 
 
 class AXIProtocolError(Exception):
+    def __init__(self,  message, xresp):
+        super().__init__(message)
+        self.xresp = xresp
+
+
+class AXIReadBurstLengthMismatch(Exception):
     pass
 
 
-class AXI4LiteMaster(BusDriver):
-    """AXI4-Lite Master.
+class AXIBurst(enum.IntEnum):
+    FIXED = 0b00
+    INCR = 0b01
+    WRAP = 0b10
+
+
+class AXIxRESP(enum.IntEnum):
+    OKAY = 0b00
+    EXOKAY = 0b01
+    SLVERR = 0b10
+    DECERR = 0b11
+
+
+class AXI4Master(BusDriver):
+    """AXI4 Master
 
     TODO: Kill all pending transactions if reset is asserted.
     """
 
-    _signals = ["AWVALID", "AWADDR", "AWREADY",        # Write address channel
-                "WVALID", "WREADY", "WDATA", "WSTRB",  # Write data channel
-                "BVALID", "BREADY", "BRESP",           # Write response channel
-                "ARVALID", "ARADDR", "ARREADY",        # Read address channel
-                "RVALID", "RREADY", "RRESP", "RDATA"]  # Read data channel
+    _signals = \
+        ["AWVALID", "AWADDR", "AWREADY", "AWID", "AWLEN", "AWSIZE", "AWBURST",
+         "WVALID", "WREADY", "WDATA", "WSTRB",
+         "BVALID", "BREADY", "BRESP", "BID",
+         "ARVALID", "ARADDR", "ARREADY", "ARID", "ARLEN", "ARSIZE", "ARBURST",
+         "RVALID", "RREADY", "RRESP", "RDATA", "RID", "RLAST"]
```

Comment:
As a general style comment, I would recommend avoiding `\` wherever possible. Here, you can do that as
```suggestion
    _signals = [
        "AWVALID", "AWADDR", "AWREADY", "AWID", "AWLEN", "AWSIZE", "AWBURST",
        "WVALID", "WREADY", "WDATA", "WSTRB",
        "BVALID", "BREADY", "BRESP", "BID",
        "ARVALID", "ARADDR", "ARREADY", "ARID", "ARLEN", "ARSIZE", "ARBURST",
        "RVALID", "RREADY", "RRESP", "RDATA", "RID", "RLAST"]
```